py3status/modules/volume_status.py

```diff
@@ -135,20 +130,107 @@ def _get_muted(self, output):
         else:
             return False
 
-    # this method is ran by py3status
+    # this method is called by py3status
     # returns a response dict
-    def current_volume(self):
+    def get_volume(self):
 
         # call amixer
-        output = check_output(shlex.split('amixer -D {} sget {}'.format(
-            self.device, self.channel))).decode('utf-8')
+        output = check_output(['amixer', '-D', self.device, 'sget', self.channel]).decode('utf-8')
 
         # get the current percentage value
         perc = self._get_percentage(output)
 
         # get info about channel mute status
         muted = self._get_muted(output)
 
+        return perc, muted
+
+    def volume_up(self, delta):
+        # volume up
+        self.run_cmd(self.cmd + ['{}%+'.format(delta)])
+
+    def volume_down(self, delta):
+        # volume down
+        self.run_cmd(self.cmd + ['{}%-'.format(delta)])
+
+    def toggle_mute(self):
+        # toggle mute
+        self.run_cmd(self.cmd + ['toggle'])
+
+
+class PulseaudioBackend(AudioBackend):
+    def __init__(self, device, channel):
+        AudioBackend.__init__(self, device, channel)
+        if self.device == 'default':
+            self.device = "0"
+        self.cmd = ["pamixer", "--sink", self.device]
```

Comment:
I like that, but at the same time, it kind of goes around the point of `__init__`.